tools/unicode_properties_parse/format_width_estimate_intervals.cpp

```diff
@@ -0,0 +1,216 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+// The following code generates data for `_Width_estimate_intervals` in <format>.
+
+#include <charconv>
+#include <format>
+#include <fstream>
+#include <iostream>
+#include <regex>
+#include <vector>
+
+void _verify(bool test, int line, const char* msg) {
+    if (!test) {
+        std::cerr << "Error at line " << line << ": " << msg << std::endl;
+        exit(EXIT_FAILURE);
+    }
+}
+#define verify(expr, msg) _verify((expr), __LINE__, (msg))
+static const char* impl_assertion_failed = "impl assertion failed";
+
+struct range_u {
+    uint32_t from, to;
+    range_u(uint32_t f, uint32_t t) : from(f), to(t) {}
+    range_u(uint32_t v) : from(v), to(v) {}
+};

```

Comment:
It seems simpler to make this an aggregate class.

On the other hand, pehaps it would be better to give it a more descriptive name like `code_point_interval` (despite being lengthy).

Reply:
Going to resolve this as it will be ultimately superseded by Python conversion.